Sources/FluentUI_iOS/Components/Pill Button Bar/SwiftUI/PillButtonBarView.swift

```diff
@@ -0,0 +1,235 @@
+    /// selecting an already selected pill will deselect it.
+    /// - Parameters:
+    ///   - style: The style of the pill buttons.
+    ///   - viewModels: The view model objects representing the pill buttons data.
+    ///   - selected: Optional generic selection binding for the pill bar.
+    ///   - centerAlignIfContentFits: If true, centrally aligns the bar if it fits the container.
+    ///   - tokenOverrides: The token overrides for the pill buttons.
+    public init(style: PillButtonStyle = .primary,
+                viewModels: [PillButtonViewModel<Selection>],
+                selected: Binding<Selection?>,
+                centerAlignIfContentFits: Bool = false,
+                tokenOverrides: [PillButtonToken: ControlTokenValue]? = nil) {
+        self.style = style
+        self.viewModels = viewModels
+        self._selected = selected
+        self.centerAlignIfContentFits = centerAlignIfContentFits
+        self.supportsPillDeselection = true
+        self.tokenOverrides = tokenOverrides
+    }
+
+    /// Initializes a new `PillButtonBarView` that restricts the bar to 1 selected pill button at all times.
+    /// - Parameters:
+    ///   - style: The style of the pill buttons.
+    ///   - viewModels: The view model objects representing the pill buttons data.
+    ///   - selected: The generic selection binding for the pill bar.
+    ///   - centerAlignIfContentFits: If true, centrally aligns the bar if it fits the container.
+    ///   - tokenOverrides: The token overrides for the pill buttons.
+    public init(style: PillButtonStyle = .primary,
+                viewModels: [PillButtonViewModel<Selection>],
+                selected: Binding<Selection>,
+                centerAlignIfContentFits: Bool = false,
+                tokenOverrides: [PillButtonToken: ControlTokenValue]? = nil) {
+        self.style = style
+        self.viewModels = viewModels
+        self.centerAlignIfContentFits = centerAlignIfContentFits
+        self.supportsPillDeselection = false
+        self.tokenOverrides = tokenOverrides
+        self._selected = Binding<Selection?>(
+            get: { selected.wrappedValue },
+            set: { newValue in
+                if let value = newValue {
+                    selected.wrappedValue = value
+                }
+            }
+        )
+    }
+
+    public var body: some View {
+        ScrollViewReader { scrollProxy in
+            ScrollView(.horizontal, showsIndicators: false) {
+                pillButtonStack()
+                    .environment(\.isEnabled, isEnabled)
+            }
+            .environment(\.isEnabled, true)
+            .scrollDisabled(disableScroll)
+            .onGeometryChange(for: CGRect.self) { proxy in
+                proxy.frame(in: .global)
+            } action: { newFrame in
+                scrollViewFrame = newFrame
+                containerWidth = newFrame.width
+            }
+            .onAppear {
+                DispatchQueue.main.async {
```

Comment:
Wherever we use a magic dispatch async, should add a comment explaining why we can't do something more deterministic in terms of timing. But natural question - can we do something more deterministic? What happens if we just do this directly?

Reply:
Since we're calling this inside onAppear, we defer to the next layout pass here to make sure scrollTo runs when scroll view's layout is fully set up. It's not strictly needed and more of a safety mechanism. I can add a comment for it.

Reply:
Nvm the new APIs are only available on iOS 17+. I can add a comment here, and a TODO to migrate to the newer APIs once iOS support is dropped in fluent.

Reply:
We could, though I don't think we should since the current API (ScrollViewReader) is wrapped around the ScrollView so we might end up with large chunks of code behind the availability checks. This would also make it harder to reason about this code if we decide to add liquid glass support which would also be behind availability checks and result in duplicated chunks of code. I think considering fluent should be dropping iOS 16 supports relatively soon we're better off keeping the current API until that happens for better code clarity. Please let me know if that makes sense, otherwise I can go ahead and make the change.

Reply:
With fluent having dropped iOS 16 support, I've migrated the APIs thanks!